core/iwasm/compilation/aot_emit_control.c

```diff
@@ -173,20 +173,56 @@ clear_frame_locals(AOTCompFrame *aot_frame)
 }
 
 static void
-restore_frame_sp(AOTBlock *block, AOTCompFrame *aot_frame)
+restore_frame_sp_for_op_else(AOTBlock *block, AOTCompFrame *aot_frame)
 {
-    uint32 all_cell_num, stack_cell_num;
+    uint32 all_cell_num =
+        aot_frame->max_local_cell_num + aot_frame->max_stack_cell_num;
+    AOTValueSlot *p_end = aot_frame->lp + all_cell_num, *p;
+
+    /* Reset all the value slots from current frame sp for the else
+       branch since they be the same as starting to translate the
+       if branch */
+    for (p = block->frame_sp_begin; p < p_end; p++) {
+        p->dirty = 0;
+        p->value = NULL;
+        p->type = 0;
+        if (aot_frame->comp_ctx->enable_gc) {
+            p->ref = 0;
+            p->committed_ref = 1;
```

Comment:
Why not reset to 0?

Reply:
`committed_ref == 1` means that the ref flag was already committed and the committed value is `0`, as `p->ref + == p->committed_ref`, no need to commit again next time.

And when translating the else branch, the status of frame refs from current frame_sp is same as translating the if branch. Note that for `if` opcode, AOT code will jump to if branch or else branch, here the `p->committed_ref` should be 1.

Reply:
Thanks!